gui/src/3D/controls/slider3D.ts

```diff
@@ -0,0 +1,193 @@
+import { Observable } from "babylonjs/Misc/observable";
+import { Vector3 } from "babylonjs/Maths/math.vector";
+import { TransformNode } from "babylonjs/Meshes/transformNode";
+import { Scene } from "babylonjs/scene";
+import { Mesh } from "babylonjs/Meshes/mesh";
+import { Control3D } from "./control3D";
+import { MeshBuilder } from "babylonjs/Meshes/meshBuilder";
+import { PointerDragBehavior } from "babylonjs/Behaviors/Meshes/pointerDragBehavior";
+import { Material } from "babylonjs/Materials/material";
+import { StandardMaterial } from "babylonjs/Materials/standardMaterial";
+import { Color3 } from "babylonjs/Maths/math.color";
+import { AbstractMesh } from "babylonjs/Meshes/index";
+
+export interface ISlider3DOptions {
+    minimum?: number,
+    maximum?: number,
+    value?: number
+};
+
+/**
+ * Class used to create a slider in 3D
+ */
+ export class Slider3D extends Control3D {
+    private _sliderBarMaterial: Material;
+    private _sliderThumbMaterial: Material;
+    private _sliderThumb: Mesh;
+    private _sliderBar: Mesh;
+
+    private _minimum: number;
+    private _maximum: number;
+    private _value: number;
+
+    /** Observable raised when the sldier value changes */
+    public onValueChangedObservable = new Observable<number>();
+
+    /**
+     * Creates a new slider
+     * @param name defines the control name
+     */
+     constructor(name?: string, options?: ISlider3DOptions) {
+        super(name);
+
+        this._minimum =  options?.minimum ? options.minimum : 0;
+        this._maximum = options?.maximum ? options.maximum : 100;
+        this.value = options?.value ? options.value : 50;       
+    }    
+
+    /** Gets or sets minimum value */
+    public get minimum(): number {
+        return this._minimum;
+    }
+
+    public set minimum(value: number) {
+        if (this._minimum === value) {
+            return;
+        }
+
+        this._minimum = Math.min(value, this._maximum);

```

Comment:
Personally I think the value should adapt to the new min/max value, instead of having the new min/max value adapt to the current value (Clamp the value to the new range, instead of expanding the range to accommodate the current value).

Reply:
Agreed, I will fix this in the next pr. Thanks

Reply:
This issue has been resolved. Take a look at the updated version.